solvers/sklearn.py

```diff
@@ -24,8 +24,15 @@ def set_objective(self, X, y, fit_intercept=False):
         warnings.filterwarnings('ignore', category=ConvergenceWarning)
 
     def run(self, n_iter):
+        if n_iter == 0:
```

Comment:
a side note: Do you see any benefit of adding a method to init stop_val for the solver to avoid having to handle that in the run method?